File: Aula03/Ex01/main.py
arr = [1, 2, 3, 4, 5]

print("Acessando lista usando índices (arr[i]):")
for i, val in enumerate(arr):
    print(f"arr[{i}] = {val}, id = {id(val)}")

print("\nEm Python não existe aritmética de ponteiros como em C.")
print("Os elementos da lista são objetos e cada id acima representa")
print("um identificador único em memória, mas não podemos incrementá-lo.")

File: Aula03/Ex01/main.c
#include <stdio.h>

int main(void) {
    int arr[5] = {1, 2, 3, 4, 5};
    int *ptr = arr; 

    printf("Acessando array usando indices (arr[i]):\n");
    for (int i = 0; i < 5; i++) {
        printf("arr[%d] = %d, endereco = %p\n", i, arr[i], (void *)&arr[i]);
    }

    printf("\nAcessando array usando ponteiro e aritmetica de ponteiros (*(ptr + i)):\n");
    for (int i = 0; i < 5; i++) {
        printf("*(ptr + %d) = %d, endereco = %p\n", i, *(ptr + i), (void *)(ptr + i));
    }

    return 0;
}